tex/appendix/ans-fo-semantics.tex

```diff
@@ -179,6 +179,24 @@ \subsection*{9.7.6}
     \emptyset$ is valid in our model.
 \end{enumerate}
 
+\subsection*{9.7.10}
+
+\begin{enumerate}
+    \item[i] We will prove $\mathcal{M} \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$ iff $P^{\mathcal{M}}$ has at least two elements. \\\\
+    Left-to right: We will prove this using proof by contradiction. Let $\mathcal{M}$ and  $\alpha$ be arbitrary such that $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. Assume $P^\mathcal{M}$ has at most 1 element. We can distinguish two cases: $P^\mathcal{M}$ has 0 elements (1) and $P^\mathcal{M}$ has 1 element (2). \\ 
+    (1) We assume $P^\mathcal{M} = \emptyset$. Then there is no $d \in D^\mathcal{M}$ such that $P(d)$ is true. This means that the conjunction $P(x) \wedge P(y) \wedge x \neq y$ can never be true. But we assumed $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. Thus, we have a contradiction. It follows that if $P^\mathcal{M} = \emptyset$, $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. \\
+    (2) We assume $P^\mathcal{M}$ has 1 element. Let this be an arbitrary element $d \in D^\mathcal{M}$. In order for $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$ to hold, it must be the case that $\mathcal{M}, \alpha_{[x \mapsto d, y \mapsto d]} \vDash (P(d) \wedge P(d) \wedge d \neq d)$. Clearly, $P(d) \wedge P(d)$ holds, but we know $d = d$. Thus, we have a contradiction. It follows that if $P^\mathcal{M}$ has one element, $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. \\\\
+    Right-to-left: We will prove this using proof by contradiction. We assume $P^\mathcal{M}$ has at least two elements and $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. The latter can be rewritten:
```

Comment:
Same as above, this is not (only) proof by contradiction, I think you could make that clearer :)

Reply:
Thanks so much for all your feedback, it has really improved the proof. It's gotten frighteningly long, but I guess that's to be expected. Hopefully I've implemented everything correctly, if not I'd be happy to make any further changes.

Reply:
Oh, I think I've correctly changed the PR to include the new proof but it's totally possible something's gone wrong again. Please let me know if you can't see it :)